smt/surrogate_models/krg_based.py

```diff
@@ -520,6 +520,24 @@ def _new_train(self):
         # if self.name != "MGP":
         #     del self.y_norma, self.D
 
+    def check_training_numerically(self):
+        """
+        Check if the training dataset could be an issue and print both
+        the dataset correlation matrix condition number and
+        minimal distance between two points.
+        """
+
+        print(
+            "Minimal distance between two points in any dimension is",
+            "{:.2e}".format(np.min(self.D)),
+        )
+        print(
+            "Correlation matrix R condition number is",
+            "{:.2e}".format(
+                np.linalg.cond(self.optimal_par["C"] @ self.optimal_par["C"])
+            ),
+        )
```

Comment:
If we could agree on some threshold values we could return a boolean indicating potential well or ill condition (maybe the function could be renamed `is_training_ill_conditioned() ?`). We could also use the `self.printer` mechanism to be able to enable/disable messages.